src/prototype_room_my.js

```diff
@@ -326,6 +326,55 @@ Room.prototype.executeRoom = function() {
     }
   }
 
+  let carryHelpInterval = config.carryHelpers.ticksUntilHelpCheck;
+  if (!Memory.needEnergyRooms) {
+    Memory.needEnergyRooms = [];
+  }
+  let key = _.indexOf(Memory.needEnergyRooms, this.name);
+
+  if (Game.time % carryHelpInterval) {
+    this.memory.energyAvailableSum += this.energyAvailable;
+
+  } else if (this.memory.energyAvailableSum === undefined) {
+    this.memory.energyAvailableSum = 0;
+
+  } else {
+    let needHelp = Memory.needEnergyRooms[key];
+    this.log(needHelp);
+    if (needHelp && this.hostile) {
+      Memory.needEnergyRooms[key] = undefined;
+      Memory.needEnergyRooms = _.compact(Memory.needEnergyRooms);
+
+    } else if (this.memory.energyAvailableSum < config.carryHelpers.needTreshold * carryHelpInterval) {
+      if (!needHelp) {
+        Memory.needEnergyRooms.push(this.name);
+        this.log('---!!!---' + this.name + ' need energy ---!!!---');
+      }
```

Comment:
Extracted to a method this could get much cleaner with `returrn`s and removing of all the nesting.